x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/snapshotlifecycle/history/SnapshotHistoryItem.java

```diff
@@ -0,0 +1,159 @@
+        this.timestamp = timestamp;
+        this.policyId = Objects.requireNonNull(policyId);
+        this.repository = Objects.requireNonNull(repository);
+        this.snapshotName = Objects.requireNonNull(snapshotName);
+        this.operation = Objects.requireNonNull(operation);
+        this.success = success;
+    }
+
+    public SnapshotHistoryItem(StreamInput in) throws IOException {
+        this.timestamp = in.readVLong();
+        this.policyId = in.readString();
+        this.repository = in.readString();
+        this.snapshotName = in.readString();
+        this.operation = in.readString();
+        this.success = in.readBoolean();
+    }
+
+    public long getTimestamp() {
+        return timestamp;
+    }
+
+    public String getPolicyId() {
+        return policyId;
+    }
+
+    public String getRepository() {
+        return repository;
+    }
+
+    public String getSnapshotName() {
+        return snapshotName;
+    }
+
+    public String getOperation() {
+        return operation;
+    }
+
+    public boolean isSuccess() {
+        return success;
+    }
+
+    @Override
+    public final void writeTo(StreamOutput out) throws IOException {
+        out.writeVLong(timestamp);
+        out.writeString(policyId);
+        out.writeString(repository);
+        out.writeString(snapshotName);
+        out.writeString(operation);
+        out.writeBoolean(success);
+        innerWriteTo(out);
+    }
+
+    @Override
+    public final XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        {
+            builder.timeField(TIMESTAMP.getPreferredName(), "timestamp_string", timestamp);
+            builder.field(POLICY_ID.getPreferredName(), policyId);
+            builder.field(REPOSITORY.getPreferredName(), repository);
+            builder.field(SNAPSHOT_NAME.getPreferredName(), snapshotName);
+            builder.field(OPERATION.getPreferredName(), operation);
+            builder.field(SUCCESS.getPreferredName(), success);
+            innerToXContent(builder, params);
+        }
+        builder.endObject();
+
+        return builder;
+    }
+
+    @Override
+    public String toString() {
+        return Strings.toString(this);
+    }
+
+    /**
+     * Write any fields that are introduced in the subclass here. This is called as part of
+     * {@link SnapshotHistoryItem#writeTo(StreamOutput)}, you do not need to (and should not) override that method. All fields that are
+     * part of the {@link SnapshotHistoryItem} class will already be written to the stream, so only write new fields introduced in the
+     * subclass.
+     * @param out The output stream
+     * @throws IOException if an error occurs while writing to the output stream
+     */
+    protected abstract void innerWriteTo(StreamOutput out) throws IOException;
```

Comment:
Rather than do this, why not have the subclass just override `writeTo` and call `super.writeTo(out)`? Do you just want to ensure that any subclass *has* to override it?

Reply:
This method is no longer relevant as I've gotten rid of the inheritance as suggested at the top level.